class/frontend-form-post.php

```diff
@@ -546,12 +580,34 @@ function submit_post() {
             }
 
             // send the response
-            $response = array(
-                'success'      => true,
-                'redirect_to'  => $redirect_to,
-                'show_message' => $show_message,
-                'message'      => $form_settings['message']
-            );
+            if ( $res_flag == false ) {
+                $response = array(
+                    'success'      => true,
+                    'redirect_to'  => $redirect_to,
+                    'show_message' => $show_message,
+                    'message'      => $form_settings['message']
+                );
+            }
+            
+            // send a unique post publish link to guest user
+
+            $post_id_encoded   = wpuf_encryption( $post_id ) ;
```

Comment:
Again, move the email preparation and sending functionalities to a separate function.